python/ray/tune/experiment/experiment.py

```diff
@@ -193,7 +193,8 @@ def __init__(
 
         self.storage = None
         if _use_storage_context():
-            assert name is not None
+            if not name:
+                name = StorageContext.get_experiment_dir_name(run)
```

Comment:
For `tune.run` usage, name may not be provided, so we need to auto-generate one. `Tuner` will always populate name though.